sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -192,11 +195,28 @@ async def send(self, request: HttpRequest, **config: Any) -> Optional[AsyncHttpR
                 allow_redirects=False,
                 **config
             )
-            response = AioHttpTransportResponse(request, result,
-                                                self.connection_config.data_block_size,
-                                                decompress=not auto_decompress)
-            if not stream_response:
-                await response.load_body()
+            if _is_rest(request):
+                from azure.core.rest._aiohttp import RestAioHttpTransportResponse
+                response = RestAioHttpTransportResponse(
+                    request=request,
+                    internal_response=result,
+                    block_size=self.connection_config.data_block_size,
+                    decompress=not auto_decompress
+                )
+                if not stream_response:
+                    try:
+                        # load_body instead of read()
+                        await response.load_body()
+                        await response.close()
+                    except Exception as exc:
+                        await response.close()
+                        raise exc
+            else:
+                response = AioHttpTransportResponse(request, result,
+                                                    self.connection_config.data_block_size,
+                                                    decompress=not auto_decompress)
+                if not stream_response:
+                    await response.load_body()
```

Comment:
Why we add this?

Seems to me this is an unnecessary behavior change.

Reply:
this behavior is the same as currently exists for the old response, it's a little hard to see, but you can see the old code on line 199 in the red. Didn't change any code for the old responses

Reply:
Get it. We just moved the code into else.

Reply:
yeah, the diff is just not showing up well in github sadly